pkg/apis/management.cattle.io/v3/catalog_types.go

```diff
@@ -288,3 +289,24 @@ type ClusterCatalog struct {
 type CatalogRefresh struct {
 	Catalogs []string `json:"catalogs"`
 }
+
+type CatalogSecrets struct {
+	CredentialSecret string `json:"credentialSecret,omitempty" norman:"nocreate,noupdate"`
+}
+
+// GetSecret gets a reference to the private catalog secret, either from the CatalogSecrets field or the Status field.
+// CatalogSecrets is preferred because Status is deprecated.
+func (c *Catalog) GetSecret() string {
+	if c.Spec.CatalogSecrets.CredentialSecret != "" {
+		return c.Spec.CatalogSecrets.CredentialSecret
+	}
+	return c.Status.CredentialSecret
+}
+
+func (c *ClusterCatalog) GetSecret() string {
+	return c.Catalog.GetSecret()
+}
+
+func (c *ProjectCatalog) GetSecret() string {
+	return c.Catalog.GetSecret()
+}
```

Comment:
Since Catalog is embedded in ClusterCatalog and ProjectCatalog these aren't necessary right?